### ISSUE TITLE
Setting anti-affinity is not working

### ISSUE THREAD

**Issue by ErlendFax** [redpanda-data/helm-charts#199]
When reading the `values.yaml` is says that one can use anti-affinity to e.g. [spread pods over multiple nodes](https://github.com/redpanda-data/helm-charts/blob/main/redpanda/values.yaml#L293), but when looking at the properties and template files it looks like this is not possible. 

My goal is to spread the pods to different nodes, if this actually is possible, please correct me.

**Comment by ErlendFax** [comment]
After chatting with some of you on slack, testing, and reading more, I am going to edit my issue a bit.

**Problem:**
When setting `podAntiAffinity.type` to `hard` or `soft` in the `values.yaml` and running `helm upgrade` it does not seem change anything.

When running `k describe pod -n redpanda redpanda-0` I don't see any changes from before, and I don't see anything about affinity or anti-affinity.

**Expectation:**
I thought changing to `hard` would change the pod deployment to mention anti-affinity somewhere.  Similar to what happens when setting node selector.

<details>

```
image:
  repository: vectorized/redpanda
  tag: v22.2.2

resources:
  cpu:
    cores: 2
    overprovisioned: true
  memory:
    enable_memory_locking: true
    container:
      min: 6.5Gi
      max: 8.0Gi

storage:
  persistentVolume:
    enabled: true
    size: 100Gi
    storageClass: "ssd-redpanda"

statefulset:
  replicas: 3
  nodeSelector:
    cloud.google.com/gke-nodepool: redpanda-pool-dev
  tolerations:
    - key: "redpanda"
      operator: "Equal"
      value: "true"
      effect: "NoSchedule"
  podAntiAffinity:
    type: hard

config:
  cluster:
    enable_idempotence: false
    enable_transactions: false

```

</details>

**Comment by joejulian** [comment]
This is the difference that changing that setting makes. Are you seeing something different?

```
$ helm template myrelease charts/redpanda --set statefulset.podAntiAffinity.type=hard >! /tmp/t1.yaml
$ helm template myrelease charts/redpanda --set statefulset.podAntiAffinity.type=soft >! /tmp/t2.yaml 
$ diff /tmp/t1.yaml /tmp/t2.yaml
340,345c340,347
<           requiredDuringSchedulingIgnoredDuringExecution:
<             - topologyKey: kubernetes.io/hostname
<               labelSelector:
<                 matchLabels:
<                   app.kubernetes.io/name: redpanda
<                   app.kubernetes.io/instance: "myrelease"
---
>           preferredDuringSchedulingIgnoredDuringExecution:
>             - weight: 100
>               podAffinityTerm:
>                 topologyKey: kubernetes.io/hostname
>                 labelSelector:
>                   matchLabels:
>                     app.kubernetes.io/name: redpanda
>                     app.kubernetes.io/instance: "myrelease"
```

**Comment by ErlendFax** [comment]
It's been a little while since I looked at this. I did not check the `helm template` as you did, but setting hard or soft did not change underlying pods YAML - which I thought would happen. No mention of anti-affinity or affinity when describing with kubectl after upgrading with helm.

I started looking at this question after redpanda went down during a cluster upgrade. I went through several settings both in the repdanda helm chart and in the GKE cluster settings. This was one of the things I couldn't verify happened the way we wanted. In the end, it was too much work for our small team so we are in the process of considering managed kafka/redpanda.
